halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ValidatorCollection.java

```diff
@@ -70,22 +70,26 @@ public int runAllValidators(ConfigProblemSetBuilder psBuilder, Node node) {
    */
   private int runMatchingValidators(ConfigProblemSetBuilder psBuilder, Validator validator, Node node, Class c) {
     int result = 0;
-
-    if (c == Node.class) {
+    
+    if (c == Object.class) {
       return result;
     }
-
+    
     try {
       Method m = validator.getClass().getMethod("validate", ConfigProblemSetBuilder.class, c);
       DaemonTaskHandler.message("Validating " + node.getNodeName() + " with " + validator.getClass().getSimpleName());
       m.invoke(validator, psBuilder, node);
       result = 1;
-    } catch (NoSuchMethodException e) {
+    } catch (InvocationTargetException | NoSuchMethodException e) {
       // Do nothing, odds are most validators don't validate every class.
-    } catch (InvocationTargetException | IllegalAccessException e) {
+    } catch (IllegalAccessException e) {
       throw new RuntimeException("Failed to invoke validate() on \"" + validator.getClass().getSimpleName() + "\" for node \"" + c.getSimpleName(), e);
     }
-
-    return runMatchingValidators(psBuilder, validator, node, c.getSuperclass()) + result;
+    
+    if (result > 0) {
```

Comment:
Given that result can be either 1 or 0 and nothing else I think a bool is more appropriate here. Returning a bool makes more sense then too.